### ISSUE TITLE
PDF Viewer not updating on st.rerun() in prod.

### ISSUE THREAD

**Issue by JanEricNitschke** [lfoppiano/streamlit-pdf-viewer#57]
I am having real trouble with an extremely weird issue. Running my application locally works fine, both in the normal poetry environment as well as the docker container. However when i host it online some functionality we use to show the next page does not cause the viewer to refresh with the next page despite changing the page that is sent to `pages_to_render` and calling a st.rerun().

However it does refresh when interacting with unrelated elements that also call `st.rerun()`. Would love to know if you have an ide where this is coming from and how to fix it.

Cheers

**Comment by lfoppiano** [comment]
Hi @JanEricNitschke, 
  thanks for reporting this issue. In order to help you I would need some more information. 

Do you have the link to the deployed application? And, do you have the code hosted somewhere? Or can you share a working example?

**Comment by JanEricNitschke** [comment]
Hi @lfoppiano , thanks for the quick response.

Unfortunately i cant show you the code or application at the moment and it hard to produce a minimal example as the issue is the discrepancy between the local and hosted version (despite docker). I also do not have much experience with streamlit. I understand that that makes it extremely hard to help/debug. I was just hoping that you might have encountered something similar before or just have a gut feeling what could be causing something like that that i could try out on my side.

**Comment by JanEricNitschke** [comment]
But i can try to describe the relevant parts a bit:

The code that displays the pdf is
```
if st.session_state[selected_pdf]:
    pdf_view = st.session_state[selected_pdf]
    display_pdf(pdf_view.content, key="some_key", pages_to_render=[pdf_view.active_page], render_text=True)
```

And then i have this to  move through the pages (this works locally but not deployed)

```
if st.session_state[selected_pdf]:
    pdf_view = st.session_state[selected_pdf]
    with col_prev:
        if st.button():
            pdf_view.active_page -= 1
            st.rerun()
    with col_num:
        def update_page_number():
            st.session_state[selected_pdf].active_page = st.session_state["temp_page"]
            st.session_state["rerun_needed"] = True

        st.number_input("somelabel", value=pdf_view.active_page, key="temp_page", on_change=update_page_number)
        if st.session_state.get("rerun_needed"):
            st.session_state["rerun_needed"] = False
            st.rerun()
```

and this to change the pdf shown:

```
with col_sel:
     for i, pdf_view in enumerate(st.session_state[üdf_kind]):
         if st.button("some_label", key="button_key")
             st.session_state[selected_pdf] = pdf_view
             st.rerun()
```

**Comment by JanEricNitschke** [comment]
I think it might be some sort of race condition as my local machine is significantly better than what I have hosted it on.

**Comment by lfoppiano** [comment]
I've released the version 0.0.15. It's not yet officially published, but it's on pypi. Could you please give it a try?

**Comment by lfoppiano** [comment]
I'm closing this issue. @JanEricNitschke feel free to reopen or comment if you still have problems.
